hardware/raspberrypi/booteeprom.md

```diff
@@ -9,36 +9,91 @@ The Raspberry Pi 4 has an SPI-attached EEPROM (4MBits/512KB), which contains cod
  - A small SPI EEPROM allows bugs to be fixed and features to be added after launch, in the field.
  - The local modifiable state means that OTP bootmode settings will not be required for PXE or USB mass storage boot on the Raspberry Pi 4. There are no user-modifiable OTP bootmode bits on Pi 4.
 
-## PXE and USB Boot
+## Network and USB boot
 
 Support for these additional bootmodes will be added in the future via optional bootloader updates. The current schedule is to release PXE boot first, then USB boot.
 
 ## Is the bootloader working correctly?
 
 To check that the bootloader is working correctly, turn off the power, unplug everything from the Raspberry Pi 4, including the SD card, and then turn the power back on. If the green LED blinks with a repeating pattern then the bootloader is running correctly, and indicating that `start*.elf` has not been found. Any other actions imply that the bootloader is not working correctly and should be reinstalled using `recovery.bin`.
 
-## recovery.bin
+## recovery.bin - rescue image
 
 If the EEPROM needs updating or has somehow become corrupted, it can be reflashed using a fresh SD card with a copy of `recovery.bin` in the first partition of an SD card, formatted to FAT32.
 
 `recovery.bin` is a special utility which runs directly from the SD card and updates the EEPROM — it is not in itself a bootloader. It flashes the green LED rapidly (forever) upon success. Because it's not a bootloader, it won't load `start*.elf`, so once you see the green LED flashing rapidly, just re-insert a regular Raspbian SD card and reboot the Pi.
 
 It can be downloaded from the [raspberrypi.org downloads page](https://www.raspberrypi.org/downloads/).
 
+## Updating the bootloader
+
+To update the bootloader for new features or bug fixes we recommend installing the rpi-eeprom Raspbian package. This installs a background service which runs at boot and checks if a critical update is available. If so, it schedules the update to be applied at the next reboot.
+
+```
+sudo apt-get update
+sudo apt-get upgrade
+sudo apt-get install rpi-eeprom
+```
+
+If you wish to control when the updates are applied you can disable the service from running automatically and run 'rpi-eeprom-update' manually (requires sudo)
+
+```
+sudo systemctl disable rpi-eeprom-update
```

Comment:
The service will run after the package is installed, so it will be too late at this point.

I think you can run `sudo systemctl mask rpi-eeprom-update` before installing.

Reply:
done